### PR TITLE
fix: react-charting VBC - RTL mode mouseover on bar chart shows incorrect data

### PR DESCRIPTION
**Pull request by harsh9975**
https://github.com/microsoft/fluentui/assets/33349461/65b0c01d-6473-4ab6-9bea-1250fcf700ad

## Previous Behavior
mouse over on the bar chart the text is displaying correctly with respect to x axis.

## New Behavior
on mouseover correct data is rendered.

- Fixes #
#29861

### CODE REVIEW

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 1]
```diff
@@ -513,7 +513,7 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
       return { xBarScale, yBarScale };
     } else {
       const xBarScale = d3ScaleBand()
-        .domain(this._xAxisLabels)
+        .domain(this._isRtl ? this._xAxisLabels.slice().reverse() : this._xAxisLabels)
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:516]: domain is not changing, range is changing.
Make change to update the range if rtl

Comment (harsh9975) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:516]: yes, fixed it

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/bab96c5f09235abc67c7c8461b9bff844bc02bad'>bab96c5f09235abc67c7c8461b9bff844bc02bad</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-charting-b1c51226-7ca4-40eb-9521-2b53b639045e.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-charting-b1c51226-7ca4-40eb-9521-2b53b639045e.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=29876&commitId=a374f5ea31de038f1a38580def6f36c806bd2d3e&env=prod&clientType=FLUENTUI)


  <details><summary><b>react-charting-VerticalBarChart</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-VerticalBarChart.Dynamic - Wrap Labels - RTL.default.chromium.png|4318| Changed|
|react-charting-VerticalBarChart.Rotated Label- Hide Legends - RTL.default.chromium.png|3324| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

Comment (AtishayMsft) [reply change/@fluentui-react-charting-b1c51226-7ca4-40eb-9521-2b53b639045e.json:1]: The PR fixes the issue reported in the baseline screenshot.

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**